lms/djangoapps/instructor/views/api.py

```diff
@@ -367,6 +384,49 @@ def register_and_enroll_students(request, course_id):  # pylint: disable=too-man
             name = student[NAME_INDEX]
             country = student[COUNTRY_INDEX][:2]
 
+            cohort = None  # Optional[CourseUserGroup]
+            course_mode = default_course_mode
+            if len(student) > COHORT_INDEX and student[COHORT_INDEX]:
+                # get a cohort name if exists, with caching in `cohorts`
+                cohort_name = student[COHORT_INDEX]
+                if not is_course_cohorted(course_id) and not already_warned_not_cohorted:
+                    row_errors.append({
+                        'username': username,
+                        'email': email,
+                        'response': _('Course is not cohorted but cohort provided. '
+                                      'Ignoring cohort assignment for all users.')
+                    })
+                    cohort = None
+                    already_warned_not_cohorted = True
```

Comment:
@Agrendalath  Agreed, it would produce repeated error messages. This could be improved with a first branch in the `if`: `if already_warned_not_cohorted` then do `cohort = None` without adding more error messages.